api/base_controller.go

```diff
@@ -55,17 +60,27 @@ type BaseController struct {
 }
 
 // NewController returns a new base controller
-func NewController(repository storage.Repository, resourceBaseURL string, objectType types.ObjectType, objectBlueprint func() types.Object, defaultPageSize, maxPageSize int) *BaseController {
+func NewController(ctx context.Context, repository storage.Repository, resourceBaseURL string, objectType types.ObjectType, objectBlueprint func() types.Object, options *Options, supportsAsync bool) *BaseController {
+	var scheduler *operations.DefaultScheduler
+	if supportsAsync {
+		scheduler = operations.NewScheduler(ctx, repository, options.OperationSettings)
+	}
+
 	return &BaseController{
 		repository:      repository,
+		scheduler:       scheduler,
 		resourceBaseURL: resourceBaseURL,
 		objectBlueprint: objectBlueprint,
 		objectType:      objectType,
-		DefaultPageSize: defaultPageSize,
-		MaxPageSize:     maxPageSize,
+		DefaultPageSize: options.APISettings.DefaultPageSize,
+		MaxPageSize:     options.APISettings.MaxPageSize,
 	}
 }
 
+func (c *BaseController) Scheduler() (bool, web.JobScheduler) {
+	return c.supportsAsync(), c.scheduler
```

Comment:
For me the scheduler first and bool second is more intuitive. Like the build in map operations.